Save editor state before reloading from disk

nikitalita opened the pull request:
<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->
Currently, the editor node will persist the current editor state to disk and subsequently reloads it when reloading a scene in [`reload_scene()`](https://github.com/godotengine/godot/blob/e45cc68092906465df3ea9e820a7c77d602addb1/editor/editor_node.cpp#L6397).  

However, when reloading from disk, `reload_scene()` isn't used, `EditorNode::_reload_modified_scenes()` is, and it does not persist the editor state when reloading from disk. This causes users to have their viewport reset and lose their place in the scene; this causes bad user experience, and would especially crop up when updating from version control.

It's a relatively simple fix; if what we're reloading is the currently edited scene, just persist the editor state to disk so that it gets reloaded when `load_scene()` is subsequently called.

The one thing that I am unsure about is whether or not we should call `apply_changes_in_editors()`; the stock editors that this is relevant to are the shader editors and the script editors, and it doesn't look like they write any changes to disk, but user plugins might write to disk and screw up the load. Please let me know your thoughts.

nikitalita committed:
Save editor state before reloading from disk

KoBeWi commented:
I don't see any difference

https://github.com/user-attachments/assets/ac653ce2-de3d-42ea-a2b9-0a098176683b

nikitalita commented:
@KoBeWi  I have pushed fixes for this. The state was reset because we didn't save the state before we started reloading (and changed tabs, etc.). 
I have changed it so that it saves the current edited scene state before we start the reload.
